src/coreclr/pal/src/misc/cgroup.cpp

```diff
@@ -401,22 +416,84 @@ class CGroup
         if (asprintf(&mem_usage_filename, "%s%s", s_memory_cgroup_path, filename) < 0)
             return false;
 
-        uint64_t temp = 0;
-        bool result = ReadMemoryValueFromFile(mem_usage_filename, &temp);
-        if (result)
+        uint64_t usage = 0;
+        bool success = ReadMemoryValueFromFile(mem_usage_filename, &usage);
+        free(mem_usage_filename);
+
+        if (!success)
+            return false;
+
+        char* stat_filename = nullptr;
+        if (asprintf(&stat_filename, "%s%s", s_memory_cgroup_path, CGROUP_MEMORY_STAT_FILENAME) < 0)
+            return false;
+
+        FILE *stat_file = fopen(stat_filename, "r");
+        free(stat_filename);
+        if (stat_file == nullptr)
+            return false;
+
+        char *line = nullptr;
+        size_t lineLen = 0;
+        size_t readValues = 0;
+        size_t active_file = 0;
+        size_t inactive_file = 0;
+        size_t dirty_file = 0;
+        char* endptr;
+
+        while (getline(&line, &lineLen, stat_file) != -1)
         {
-            if (temp > std::numeric_limits<size_t>::max())
+            if (strncmp(line, s_active_file_key_name, s_active_file_key_length) == 0)
```

Comment:
A tiny nit: this string comparison will incorrectly evaluate to true if `line` begins with contents of `s_active_file_key_name` but contains additional characters. [More here](http://www.cpp.re/forum/beginner/109868/3/).

Reply:
@omajid It is actually my intention to use `strncmp`  as a prefix match. Lines in the `memory.stat` file are in a form of:
`<key_name><space><value>`. In my code I'm treating the space character as integral part of the name of a key. This is to avoid any mistakes in case when a `key_name` is a prefix of another `key_name` .